### ISSUE TITLE
role_id with more than 30 characters doesn't match regexp

### ISSUE THREAD

**Issue by pdecat** [terraform-providers/terraform-provider-google#3658]
### Terraform Version

```
# terraform version
Terraform v0.11.14
+ provider.google v2.6.0
+ provider.google-beta v2.6.0
```

### Affected Resource(s)
Please list the resources as a list, for example:
- `google_project_iam_custom_role`

### Terraform Configuration Files
```hcl
resource "google_project_iam_custom_role" "storage_object_creator" {
  role_id     = "myproject.cloudStorageObjectCreator"
  title       = "My Project Cloud Storage Object Creator"
  description = "Role to enable object creation access to a Google Cloud Storage Bucket"

  permissions = [
    "storage.objects.create",
  ]
}
```

### Debug Output

N/A

### Panic Output

N/A

### Expected Behavior
What should have happened?

Configuration should apply fine like in previous versions up to 2.5.1.

### Actual Behavior
What actually happened?

Configuration is rejected:
```
Error: google_project_iam_custom_role.storage_object_creator: "role_id" ("myproject.cloudStorageObjectCreator") doesn't match regexp "^[a-zA-Z0-9_\\.\\-]{1,30}$"
```

### Steps to Reproduce

1. `terraform plan`

### Important Factoids

N/A

### References

N/A

**Comment by pdecat** [comment]
This is caused by https://github.com/terraform-providers/terraform-provider-google/commit/bc232a21aadcec78ab3882730757c599d0943164

**Comment by pdecat** [comment]
This role id is accepted by gcloud:

```
# gcloud --project myproject iam roles create myproject.cloudStorageObjectCreator --description "Role to enable object creation access to a Google Cloud Storage Bucket" --title "My Project Cloud Storage Object Creator" --permissions storage.objects.create
Created role [myproject.cloudStorageObjectCreator].
description: Role to enable object creation access to a Google Cloud Storage Bucket
etag: BwWJToUtwUE=
includedPermissions:
- storage.objects.create
name: projects/myproject/roles/myproject.cloudStorageObjectCreator
stage: ALPHA
title: My Project Cloud Storage Object Creator
```
```
# gcloud --project myproject iam roles describe myproject.cloudStorageObjectCreator
description: Role to enable object creation access to a Google Cloud Storage Bucket
etag: BwWJToUtwUE=
includedPermissions:
- storage.objects.create
name: projects/myproject/roles/myproject.cloudStorageObjectCreator
stage: ALPHA
title: My Project Cloud Storage Object Creator
```

**Comment by rremer** [comment]
"myproject.cloudStorageObjectCreator" is 36 characters, while the documentation the validation function was written from specifies a max of 30 characters. I'll find out what the actual boundary is and update the documentation and validation function.

For reference: https://cloud.google.com/iam/docs/understanding-custom-roles#naming_the_role

**Comment by pdecat** [comment]
Here's the actual limit:

```
# gcloud --project myproject iam roles create testrole0123456789012345678901234567890123456789012345678901234567890123456789 --description "Role to enable object creation access to a Google Cloud Storage Bucket" --title "My Project Cloud Storage Object Creator" --permissions storage.objects.create
ERROR: (gcloud.iam.roles.create) INVALID_ARGUMENT: The role_id "testrole0123456789012345678901234567890123456789012345678901234567890123456789" is invalid. It doesn't match pattern "[a-zA-Z0-9_\.]{3,64}". The role_id must be 3 to 64 characters long and can be a mix of uppercase and lowercase English letters, digits, underscores and periods.
```

**Comment by danawillow** [comment]
Thanks for reporting, @pdecat! Next time, would you mind keeping the lines in the issue template that say to keep them for our automation? That helps HashiBot automatically assign the correct label to the issue, which makes it easier for us to triage. It's also nice to keep the community note too that way people understand how we hope they react to the issue.

**Comment by pdecat** [comment]
Hi @danawillow, I was actually surprised not to find the usual issue template when I opened this issue this morning.
When I clicked on the "New issue" button, I first had a github error message and I when I refresh, I had the default issue template instead: 
![image](https://user-images.githubusercontent.com/318490/58051625-fcc7d480-7b52-11e9-860a-78d27f28f739.png)

**Comment by pdecat** [comment]
Actually, I had an error when accessing https://github.com/terraform-providers/terraform-provider-google/issues/new/choose, refreshing did not help, then I removed the `/choose` part from the URL and it worked.

Maybe the default template should also be updated.

**Comment by danawillow** [comment]
Ah I see, thanks! Will do!

**Comment by pdecat** [comment]
Thanks @rremer & @danawillow!

**Comment by pdecat** [comment]
Fix confirmed with provider version 2.7.0, thanks again!
